server/plugins/utils/fma_fetch_album.py

```diff
@@ -0,0 +1,115 @@
+        },
+    ).drop_duplicates()
+
+
+    tmp_path = images_path / 'tmp'
+    tmp_path.mkdir(exist_ok=True, parents=True)
+
+
+    client = dc.Client(
+        '<ADD_USER_AGENT>',
+        user_token='<ADD_DISCOGS_USER_TOKEN>'
+    )
+
+
+    for __, row in album_artist_df.iterrows():
+
+        # Album cover is already downloaded
+        if len(list(images_path.glob(f"{row['album_id']}.*"))) > 0:
+            # print('exists') 
+            continue
+
+        print(f"Downloading: {row['album_name']} by {row['artist']}")
+        
+        # Download
+        try:
+            results = client.search(
+                row['album_name'],
+                type='release',
+                artist=row['artist'],
+                per_page=1,
+                page=1
+            )
+
+        
+            thumbnail_url = results[0].thumb
+            response = requests.get(thumbnail_url, headers={'user-agent': '<ADD-AGENT-NAME>'})
+
+            if response.status_code != 200:
+                raise URLError(f'Status code = {response.status_code}')
+            
+            extension = imghdr.what(file=None, h=response.content)
+            save_path = f"{row['album_id']}.{extension}"
+            with open(images_path/save_path, 'wb') as f:
+                f.write(response.content)
+
+        except Exception as e :
+            print(f"Download failed because [{type(e)}]: {e}")
+            print(f'Fallback to artist {row["artist"]}')
+        
+            save_path = f"{row['album_id']}.none"
+            (images_path/save_path).touch()
+
+            if len(list(images_path.glob(f"artist_{row['artist_id']}.*"))) > 0:
+                continue
+
+            try:
+                results = client.search(
+                    row['artist'],
+                    type='artist',
+                    per_page=1,
+                    page=1
+                )
+            
+                image = results[0].images[0]
+                if 'uri150' in image:
+                    thumbnail_url = results[0].images[0]['uri150']
+                else:
+                    thumbnail_url = results[0].images[0]['resource_url']
+
+                response = requests.get(thumbnail_url, headers={'user-agent': '<ADD_USER_AGENT>'})
+
+                if response.status_code != 200:
+                    raise URLError(f'Status code = {response.status_code}')
+                
+                extension = imghdr.what(file=None, h=response.content)
+                save_path = f"artist_{row['artist_id']}.{extension}"
+                with open(images_path/save_path, 'wb') as f:
+                    f.write(response.content)
+
+            except Exception as e :
+                print(f"Fallback failed because [{type(e)}]: {e}")
+                save_path = f"artist_{row['artist_id']}.none"
+                (images_path/save_path).touch()
+
+
+
+if __name__ == '__main__':
+    img_data_path = Path('server/static/datasets/fma/img')
```

Comment:
Potenciálně problematické, fungovalo Vám to bez problémů? Čekal bych, že když spustíte tento skript z adresáře kde se nachází, tak to tu cestu nenajde. Relativně snadný fix (ikdyž možná ošklivý) by mohlo být nahrazení za
```
dataset_path = os.path.join(get_abs_project_root_path(), 'static', 'datasets')
img_data_path = Path(f'{dataset_path}/fma/img')
tracks_path = Path(f'{dataset_path}/fma/tracks.csv')
```
čímž získáte abs cestu k "server/" adresáři místo aktuálně použité relativní